.travis.yml

```diff
@@ -1,51 +1,43 @@
 # doc: http://about.travis-ci.org/docs/user/languages/scala/
-#
-# Java flags reasoning, also see https://weblogs.java.net/blog/kcpeppe/archive/2013/12/11/case-study-jvm-hotspot-flags
-#
-# -XX:+CMSClassUnloadingEnabled - effectively adds GC to Perm space.
-# -XX:+UseConcMarkSweepGC       - must be enabled for CMSClassUnloadingEnabled to work
-# -XX:MaxPermSize=300M          - In Java 8, this setting is non-mutable and set at 300M - let's be prepared. Also, for Java 7, 
-#                                 fixing the actual size is not recommended - see above article 
-# 
-# SBT Concurrency: 
-# By default, SBT runs tasks concurrently and sets the limit of parallel tasks by looking at the number of cores available.
-# Note however, that each tasks uses more memory as well, so machines with more cores will need to allocate more memory to SBT.
-# On Travis, there is currently 2 cores and so the default SBT java settings are adequate. The rule seems to be to set
-# the flag -XmxN, where N = No. cores / 4 - eg with 8 cores(4 real, with hyper threading 8) - -Xmx2048m
-#
-
-# use Docker
-# see http://blog.travis-ci.com/2014-12-17-faster-builds-with-container-based-infrastructure/
+
 sudo: false
 
 language: scala
 
-env:
-  export SBT_OPTS="-XX:+UseConcMarkSweepGC -XX:+CMSClassUnloadingEnabled -XX:MaxPermSize=300M -XX:ReservedCodeCacheSize=128m -Xmx2G"
+cache:
+  directories:
+    - $HOME/.m2/repository
+    - $HOME/.sbt
+    - $HOME/.ivy2
+
+script:
+  - sbt ++$TRAVIS_SCALA_VERSION -jvm-opts travis/jvmopts test
+  # Tricks to avoid unnecessary cache updates
+  - find $HOME/.sbt -name "*.lock" | xargs rm
```

Comment:
IMHO, this is the opposite of what is required. A clean cache - ie no cache - will check that all required dependencies are brought in by _this_ build, rather than someone else's. eg: your build works, you push code, someone else pulls and the build fails. So, you delete your cache, rebuild and fails, so you add missing deps...